tensorboard/compat/tensorflow_stub/io/gfile.py

```diff
@@ -140,6 +156,10 @@ def listdir(self, dirname):
         entries = [compat.as_str_any(item) for item in entries]
         return entries
 
+    def makedirs(self, path):
+        """Creates a directory and all parent/intermediate directories."""
```

Comment:
`os.makedirs()` has the following behavior: "Raises an error exception if the leaf directory already exists or cannot be created."

That should be documented here, and the S3 version should maintain the same behavior contract - does it?  If it's difficult for S3 to do that, we could choose a different behavior (like "has no effect if the leaf directory already exists") but then we should still document it and ensure it's consistent across FS implementations.